### DISCUSSION TITLE
Flutter Web App with nested routing not starting and throws: bad HTTP response code (404) was received when fetching the script

### DISCUSSION THREAD

**Discussion by JonasJW** [vercel/vercel#8861] [Help]
I have been struggling with an issue on my flutter web application that is deployed on vercel for a while. I'm assuming this issue is related to vercel but I can't exclude it being a flutter issue. I would really appreciate any help on this because it has been keeping me guessing for a while now. I have also [asked on Stackoverflow](https://stackoverflow.com/questions/74305233/flutter-web-app-not-starting-the-script-has-an-unsupported-mime-type) before.

## How to reproduce

I have created a demo flutter web application here: https://github.com/JonasJW/vercel_flutter_web_test

To deploy it to vercel (which I already did [here](https://vercel-flutter-web-test.vercel.app/)), one must select the Framework `other`, set install command to `if cd flutter; then git pull && cd .. ; else git clone https://github.com/flutter/flutter.git; fi && ls && flutter/bin/flutter doctor && flutter/bin/flutter clean && flutter/bin/flutter config --enable-web` and build command to `flutter/bin/flutter build web`.

## Further explanation

It is a very simple application that uses a nested router with one route being `/home/questionnaire`. Because it is a SPA, I need to setup vercel's rewrites. My current `vercel.json` looks like this:

    {
        "rewrites": [
            {
                "source": "/home/questionnaire",
                "destination": "/"
            },
            {
                "source": "/home",
                "destination": "/"
            }
        ]
    }

Originally I wanted to setup my rewrites like this:

    {
        "rewrites": [
            {
                "source": "/:a*",
                "destination": "/"
            },
            {
                "source": "/home/questionnaire",
                "destination": "/"
            }
        ]
    }

But here I get a similar behavior (But some other errors in the console).

## behavior and the problem

The app deploy just fine to vercel. I can open `https://vercel-flutter-web-test.vercel.app` and I can open `https://vercel-flutter-web-test.vercel.app/home` and everything works fine. But when I open `https://vercel-flutter-web-test.vercel.app/home/questionnaire`, I get the error ` bad HTTP response code (404) was received when fetching the script.` in the console and it shows that the flutter scripts are not able to execute, so the app does not start.

The reason I assume this to be a vercel issue is that I don't get this behavior on other services, such as Firebase hosting.

Any help regarding this would be highly appreciated

**Comment by amyegan** [comment]
Hi @JonasJW! It looks like `/home` loads fine in your example, but when I visit `/home/questionnaire` it seems to log successfully from `questionnaire` before logging these errors and warnings in the browser console

<img width="393" alt="image" src="https://user-images.githubusercontent.com/8247105/200432712-e055466c-3cbb-49ef-a56a-037ee67d44a8.png">

I'm not very familiar with Flutter, so please forgive me for asking dumb questions! Is the `/home/flutter_service_worker.js` script supposed to be generated at build time? I don't see it in the `/web` directory but maybe it's coming from somewhere else that I just don't see right now.

I do see the deployed example happily logging "LOAD!" from [/web/index.html line 45](https://github.com/JonasJW/vercel_flutter_web_test/blob/c58d17dffd75064b151fe0c04004f00bcddd65f6/web/index.html#L45) before it throws a 404 loading the service worker. Maybe the `$FLUTTER_BASE_HREF` [environment variable](https://vercel.com/docs/concepts/projects/environment-variables) needs to be modified as suggested in this [Stack Overflow post](https://stackoverflow.com/a/69966604).

**Reply by JonasJW** [reply]
Hi @amyegan, thanks for looking at this issue.

Exactly, this error only occurs with nested routes deeper than the first level.

Not a stupid question at all. Flutter generates everything web related itself, so the `index.html` is generated when creating a flutter project. When building the app, flutter compiles everything to `build/web` where it also generates the `flutter_service_worker.js ` file.

However, as you pointed out, it tries to find the service worker at `/home/flutter_service_worker.js`, while it is located at `/flutter_service_worker.js`. As in the linked Stackoverflow post, the warning seems to be resolved when using `<base href="./">`. But the actual error `Failed to load resource: the server responded with a status of 404` still persists.

But I have found a workaround. When I use a different `index.html` that was generated from an older flutter version, [see here](https://github.com/JonasJW/vercel_flutter_web_test/blob/test/web/index.html), it seems to work. I still don't understand what caused the error or if it is related to Flutter or Vercel.

Thanks for your help.

**Reply by amyegan** [reply]
Interesting. I'm glad you found a workaround! I don't know many people using Flutter on Vercel, but I'll ask around to try to figure out what's going wrong with the newer version of Flutter on Vercel.

**Reply by JonasJW** [reply]
Great, thank you! Hopefully, more people use Flutter web with Vercel so it eventually gets supported as a framework for easier deployment :)
